### DISCUSSION TITLE
Trouble showing Region column

### DISCUSSION THREAD

**Discussion by willinrecovery** [code4recovery/12-step-meeting-list#1373] [Q&A] [answered]
I have configured Regions for every city in our Intergroup and associated every meeting with a Region. 

However, the Region column is not showing up for the meeting list even though I see it showing up on other sites using the 12-Step Meeting List plugin. Is there something else I need to do to activate that column and the corresponding filter in our meeting list?

Here is our meeting list:

<img width="1143" alt="Monosnap CECDA Staging 0 2024-03-03 15-19-50" src="https://github.com/code4recovery/12-step-meeting-list/assets/159723102/b3300b1d-84d7-4be3-b5e3-bc40fee7c1ab">

Here is the meeting list from another site using the same plugin. I notice it also says "Address / Platform" instead of just "Address" so I am wondering if I have missed a setting somewhere that turns these things on.

<img width="1316" alt="Monosnap Meetings 2024-03-03 15-22-10" src="https://github.com/code4recovery/12-step-meeting-list/assets/159723102/b826f9aa-3583-4690-a5d0-794989f72470">

Thank you so much!

**Comment by SIA-WEBSUP** [comment]
You should see the REGIONS dropdown. 

Can give us the link to your page?
   should be https://yourwebsite.org/meetings by default

Can you give us a link to the JSON feed?
  should be  https://yourwebsite.org/wp-admin/admin-ajax.php?action=meetings by default

**Reply by SIA-WEBSUP** [reply]
I just spun up a test site and even after adding a single meeting the REGION dropdown showed up correctly.

**Reply by SIA-WEBSUP** [reply]
Would you change the UI to Legacy and see if this changes anything? It may be specific to TSML-UI.

![image](https://github.com/code4recovery/12-step-meeting-list/assets/63924390/664d0733-6861-42c5-8f0e-1d0f2647e8f9)

**Reply by willinrecovery** [reply]
First, here is the link to the page: https://staging.debtorsanonymous.ca/meetings/

I am guessing the JSON feed would be https://staging.debtorsanonymous.ca/wp-admin/admin-ajax.php?action=meetings I haven't changed anything in relationship to the feed that I am aware of. I cannot even find where to configure that. I do see a setting to make the feed open or restricted. It is currently set to restricted.

In terms of the User Interface Display, it was already set to Legacy. I just changed it to TSML-UI to see if that might resolve the problem. It did not.

Thank you so much for looking at this.

**Reply by SIA-WEBSUP** [reply]
Please make the feed open.

**Reply by willinrecovery** [reply]
Feed in now open.

**Comment by joshreisner** [comment]
i think the issue here may be that TSML UI doesn't show a filter when all the options are the same - do you have different regions set on different meetings?

**Reply by joshreisner** [reply]
hm, i see the feed now. i see that the top-level parent region is always the same. i think if you take that away that you will see the regions.

i'll need to look at TSML UI more to understand what we want to have happen in this case

**Reply by willinrecovery** [reply]
You are correct in terms how I had configured the Regions. I had all of the provinces for this Intergroup under one category for the whole intergroup. I changed that.

Unfortunately, even though I now have each of the provinces as completely separate Regions and the Regions column and Regions filter are still not showing.

**Reply by joshreisner** [reply]
in your JSON feed, i am still seeing `Central and Eastern Canada` as the top level region

**Reply by willinrecovery** [reply]
I have updated the hierarchy of the Regions.  Is there something else I need to do to update the JSON feed?

<img width="804" alt="Monosnap CECDA Staging (admin) 2024-03-04 11-41-49" src="https://github.com/code4recovery/12-step-meeting-list/assets/159723102/b1b978a7-f43f-4458-be4d-42cdc8eec813">

**Reply by joshreisner** [reply, marked as answer]
hm, you might try editing and saving changes on an individual meeting. i think this will repopulate the JSON feed

**Comment by SIA-WEBSUP** [comment]
@willinrecovery, the screenshots you originally shared show TSML-UI and not the legacy UI.  Currently, your site is set to TSML-UI.

I would be curious to see what happens if you switch to Legacy UI.

When I import your feed into my test sight, I can see the regions and subregions correctly:
![image](https://github.com/code4recovery/12-step-meeting-list/assets/63924390/145d5148-6d2b-44af-a44e-fed3dd4e8e4e)

**Comment by SIA-WEBSUP** [comment]
Hmmm... in my test site where I'm importing your feed TSML UI behaves OK too.
![image](https://github.com/code4recovery/12-step-meeting-list/assets/63924390/fc86748b-9a8c-4110-9ba3-f03ae1b1e3ef)

**Reply by joshreisner** [reply]
the issue is that @willinrecovery has three levels of region hierarchy, but when you import to TSML it only knows how to import the bottom two - this is why it's working on your site and not on the origin

if @willinrecovery removes the top level, which is always the same, then it will look like this screenshot

**Reply by willinrecovery** [reply]
Unfortunately, I tried that, but the Regional column and the Regional filter are still not showing.

As to the interface, when I first reported the problem, it was set to Legacy. I tried setting to TSML-UI attempting to resolve the problem but it made no difference. It is still set to TSML-UI.

**Reply by SIA-WEBSUP** [reply]
I don't have a ton of experience with nested regions, but my suspicion is that ( internally anyhow ) the region hierarchy is still three deep with the top region being "Central and Eastern Canada".

If you take a look at the JSON feed, you'll find that every meeting's regions array is three deep.
    https://staging.debtorsanonymous.ca/wp-admin/admin-ajax.php?action=meetings

![image](https://github.com/code4recovery/12-step-meeting-list/assets/63924390/b7f1879f-15c4-4f38-8c69-40b0a0eb734d)

**Reply by willinrecovery** [reply]
There must be some kind of caching going on. That is the way I had set up the Regions, but I have edited them so that the provinces are no longer under "Central and Eastern Canada". At least in terms of the Individual Regions.

<img width="804" alt="Monosnap CECDA Staging (admin) 2024-03-04 11-41-49" src="https://github.com/code4recovery/12-step-meeting-list/assets/159723102/4c7ef124-6e64-43df-97b9-e8f01e8ef1c8">

**Reply by SIA-WEBSUP** [reply]
That does look like it would do it the trick. It may be a bug involving 'un-nesting' a region, but I can't be sure without getting a closer look.

Two things that occur to me that you might consider doing if you wanted to experiment further : 
   ( be sure to back up your website if you wanted to get back to where you are currently ):
    1. what happens if you delete the "Central and Eastern Canada" region?
    2. what happens if you download the meetings as a CSV, delete all meetings, and them import the CSV?

**Comment by SIA-WEBSUP** [comment]
Oh, hey... the REGION dropdown is now showing up correctly on your site.
Did you do something since your last post that fixed the issue?

**Reply by willinrecovery** [reply]
Yes, thank you. Sorry for not responding sooner.

I edited two or three meetings and saved them. I didn't need to edit the whole list.

When I refreshed the page, the Region column and drop-down appeared as you see now.

I could swear someone on here suggested I try this, but I cannot find the comment to mark it as the answer.

Thank you so much for your service!!

**Reply by willinrecovery** [reply]
Found the answer and marked it. Thank you again!
